tests/test_rapidast_run_scanner.py

```diff
@@ -0,0 +1,50 @@
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import rapidast
+from rapidast import scanners
+
+
+@patch("rapidast.scanners.str_to_scanner")
+def test_run_scanner_setup_failure(mock_str_to_scanner):
```

Comment:
Could you add some brief comments to these two tests explaining the what they are testing? I don't think they are, but whenever I see methods like `assert_called()` used I start to suspect the tests might be [change detector tests](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html)

Reply:
Added comments. 

Regarding change detector tests, I don't think these two tests qualify as such. The purpose of `run_scanner` is to execute a series of methods in sequence and return either 0 or 1. The simplest way to test `run_scanner`  behavior, based on the `scanner.setup` method, is to mock all dependencies and verify the mocks for both success and failure. While we could write an integration test without mocking, it would be more complex.

Any suggestions for improving these tests would be appreciated!